app/views/articles/index.html.erb

```diff
@@ -38,41 +38,42 @@
-            </li>
-            <li>
-              <a data-text="Week" href="<%= list_path %>/top/week" class="crayons-tabs__item <%= "crayons-tabs__item--current" if timeframe_check("week") %>"<%= timeframe_check("week") ? ' aria-current="page"'.html_safe : "" %>>Week</a>
-            </li>
-            <li>
-              <a data-text="Month" href="<%= list_path %>/top/month" class="crayons-tabs__item <%= "crayons-tabs__item--current" if timeframe_check("month") %>"<%= timeframe_check("month") ? ' aria-current="page"'.html_safe : "" %>>Month</a>
-            </li>
+      <header class="p-2 px-3 m:p-0 m:px-0 m:mb-2 fs-l">
+        <h1 class="screen-reader-only">Posts</h1>
+        <nav class="-mx-3 m:mx-0 s:flex items-center justify-between" aria-label="View posts by">
+          <ul class="crayons-navigation crayons-navigation--horizontal">
             <li>
-              <a data-text="Year" href="<%= list_path %>/top/year" class="crayons-tabs__item <%= "crayons-tabs__item--current" if timeframe_check("year") %>"<%= timeframe_check("year") ? ' aria-current="page"'.html_safe : "" %>>Year</a>
+              <a data-text="Feed" href="<%= list_path %>/" class="crayons-navigation__item <%= "crayons-navigation__item--current" if %w[week month year infinity latest].exclude?(params[:timeframe]) %>"<%= %w[week month year infinity
+                                                                                                                                                                                                                 latest].exclude?(params[:timeframe]) ? ' aria-current="page"'.html_safe : "" %>>Feed</a>
             </li>
             <li>
-              <a data-text="Infinity" href="<%= list_path %>/top/infinity" class="crayons-tabs__item <%= "crayons-tabs__item--current" if timeframe_check("infinity") %>"<%= timeframe_check("infinity") ? ' aria-current="page"'.html_safe : "" %>>Infinity</a>
+              <a data-text="Latest" href="<%= list_path %>/latest" class="crayons-navigation__item <%= "crayons-navigation__item--current" if timeframe_check("latest") %>"<%= timeframe_check("latest") ? ' aria-current="page"'.html_safe : "" %>>Latest</a>
             </li>
             <li>
-              <a data-text="Latest" href="<%= list_path %>/latest" class="crayons-tabs__item <%= "crayons-tabs__item--current" if timeframe_check("latest") %>"<%= timeframe_check("latest") ? ' aria-current="page"'.html_safe : "" %>>Latest</a>
+              <a data-text="Top" href="<%= list_path %>/top/week"
+                class="crayons-navigation__item <%= "crayons-navigation__item--current" if timeframe_check("week") || timeframe_check("month") || timeframe_check("year") || timeframe_check("infinity") %>"
+                <%= timeframe_check("week") || timeframe_check("month") || timeframe_check("year") || timeframe_check("infinity") ? ' aria-current="page"'.html_safe : "" %>>
+                  Top
+              </a>
```

Comment:
From an accessibiity / semantic HTML POV this would make more sense as a button trigger for expandable content (the set of timeframe filters) rather than a link in itself (since there is no "Top" page - it's going to "top/week").

Not sure how'd you'd feel about that from the design side of things - is it essential that when we click on "Top" we go to a new page? Or could we refactor so that clicking "Top" just expands the actual links for each of the timeframes?

Reply:
I'm actually leaning towards current implementation: so clicking "Top" takes you to /top/week. But here's my rationale:

1. That feels like expected behavior: as other links in tabs take you somewhere I feel like this could too.
  - if we ever make everything SPA and async load we could do both probably - preload `/top/week` and open dropdown. But I don't think it's worth the effort today.
2. I think this is more standard and common pattern - I'm just looking at for example reddit (they have similar concept for "Top") as well as Facebook's search results (and filtering on the left) – that initial tap always takes you somewhere else except just showing dropdown options.
3. It's faster. For those who actually want to see Top Week, it's just one less tap.
4. It also simply won't work with this design (UI wise) as we're showing these extra options (week, month, ..) on the opposite side of this area. If we do that with JS (async, without changing context like page reloading) it will either be weird or require changing design to something completely different.